src/System.Net.Sockets/tests/FunctionalTests/KeepAliveTest.netcoreapp.cs

```diff
@@ -0,0 +1,67 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading;
+using Xunit;
+
+namespace System.Net.Sockets.Tests
+{
+    public class KeepAliveTest : IDisposable
+    {
+        private const bool enabled = true;
+        private const int retryCount = 60;
+        private const int time = 5;
+        private const int interval = 2;
+
+        private readonly Socket socket;
+        
+        public KeepAliveTest()
+        {
+            socket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
+        }
+
+        [Fact]
+        public void Socket_KeepAlive_Disabled_By_Default()
+        {
+            Assert.False(IsKeepAliveEnabled(socket), "Keep-alive was turned on by default!");
+        }
+
+        [Fact]
+        public void Socket_KeepAlive_Enable_Success()
+        {
+            socket.SetSocketOption(SocketOptionLevel.Socket, SocketOptionName.KeepAlive, true);
+            Assert.True(IsKeepAliveEnabled(socket));
+        }
+
+        [Fact]
+        [PlatformSpecific(TestPlatforms.AnyUnix)]
```

Comment:
Why is it Unix specific? We need the test also on Windows (we just need to adjust expected value of retry-count when it is not supported on specific Windows version ...

Reply:
Do you want tests to be like if Windows 1703/1709 test features supported by that OS+version otherwise do nothing (go green)?

Reply:
Because both `SetSocketOption` and `GetSocketOption` are supported starting from Windows 1703 or 1709 for RetryCount, Time and Interval and I didn't want to have tests that run on all OSes/versions but do different things
Anyway I will add an if in tests to return when the OS/version is not supported